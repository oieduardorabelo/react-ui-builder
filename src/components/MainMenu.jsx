import React from 'react'
import { DragSource } from 'react-dnd'

const menuItemSource = {
  beginDrag(props) {
    return {
      name: props.name.toLowerCase().replace(/\s/g, ''),
    }
  },
  endDrag(props, monitor) {
    const item = monitor.getItem()
    const dropResult = monitor.getDropResult()
    if (dropResult) {
      console.log(`You dropped ${item.name} into ${dropResult.name}!`)
    }
  },
}

@DragSource('Section', menuItemSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging(),
}))
class MenuItem extends React.Component {
  static propTypes = {
    name: React.PropTypes.string.isRequired,
    connectDragSource: React.PropTypes.func.isRequired,
    isDragging: React.PropTypes.bool.isRequired,
  }

  render() {
    const { isDragging, connectDragSource, name } = this.props
    return connectDragSource(
      <li className="MainMenuItem" style={{ opacity: isDragging ? 0.5 : 1 }}>
        <span className="MainMenuItemText">{name}</span>
      </li>
    )
  }
}

const MainMenu = () => (
  <ul className="MainMenu">
    <MenuItem name="Section 1" />
    <MenuItem name="Section 2" />
    <MenuItem name="Section 3" />
  </ul>
)

export default MainMenu
