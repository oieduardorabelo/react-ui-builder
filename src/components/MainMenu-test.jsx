import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import MainMenu from './MainMenu'

describe('<MainMenu />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MainMenu onClickMenuItem={() => {}} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
