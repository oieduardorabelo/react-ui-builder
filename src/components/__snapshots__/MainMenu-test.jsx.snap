// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> renders correctly 1`] = `
<ul
  className="MainMenu"
>
  <DragSource(MenuItem)
    name="Section 1"
    onClickMenuItem={[Function]}
  />
  <DragSource(MenuItem)
    name="Section 2"
    onClickMenuItem={[Function]}
  />
  <DragSource(MenuItem)
    name="Section 3"
    onClickMenuItem={[Function]}
  />
</ul>
`;
